internal/variants/variants.go

```diff
@@ -71,8 +81,29 @@ func (q *Query) ValidateInput() error {
 	if q.Allele == "" {
 		return errors.New("missing allele")
 	}
-	if q.Coord == nil {
-		return errors.New("missing coordinate")
+	if err := q.validateCoordinates(); err != nil {
+		return fmt.Errorf("validating coordinates: %v", err)
+	}
+	return nil
+}
+
+func (q *Query) validateCoordinates() error {
+	var precisePosition, imprecisePosition bool
+	if q.Start != nil && (q.End != nil || q.Allele != "") {
+		precisePosition = true
+	}
+	if q.StartMin != nil && q.StartMax != nil && q.EndMin != nil && q.EndMax != nil {
+		imprecisePosition = true
+	}
+
+	if precisePosition && imprecisePosition {
+		return errors.New("please query either precise or imprecise position")
```

Comment:
errors.New("either precise or imprecise positions may be specified, not both")

Reply:
done